src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -1480,3 +1482,243 @@ void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *
 }
 
 
+//
+// QgsFieldDomainItemGuiProvider
+//
+
+QString QgsFieldDomainItemGuiProvider::name()
+{
+  return QStringLiteral( "field_domain_item" );
+}
+
+QWidget *QgsFieldDomainItemGuiProvider::createParamWidget( QgsDataItem *item, QgsDataItemGuiContext )
+{
+  if ( QgsFieldDomainItem *fieldDomainItem = qobject_cast< QgsFieldDomainItem * >( item ) )
+  {
+    const QgsFieldDomain *domain = fieldDomainItem->fieldDomain();
+    return new QgsFieldDomainDetailsWidget( nullptr, domain );
+  }
+  else if ( QgsFieldDomainsItem *fieldDomainsItem = qobject_cast< QgsFieldDomainsItem * >( item ) )
+  {
+    return new QgsFieldDomainsDetailsWidget( nullptr, fieldDomainsItem->providerKey(), fieldDomainsItem->connectionUri() );
+  }
+  else
+  {
+    return nullptr;
+  }
+}
+
+//
+// QgsFieldDomainDetailsWidget
+//
+
+QgsFieldDomainDetailsWidget::QgsFieldDomainDetailsWidget( QWidget *parent, const QgsFieldDomain *domain )
+  : QWidget( parent )
+{
+  setupUi( this );
+
+  mDomain.reset( domain->clone() );
+
+  const QString style = QgsApplication::reportStyleSheet();
+  mTextBrowser->document()->setDefaultStyleSheet( style );
+
+
+  QString metadata = QStringLiteral( "<html>\n<body>\n" );
+  metadata += QStringLiteral( "<h1>" ) + domain->name() + QStringLiteral( "</h1>\n<hr>\n" ) + QStringLiteral( "<table class=\"list-view\">\n" );
+
+  metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Description" ) + QStringLiteral( "</td><td>" ) + domain->description() + QStringLiteral( "</td></tr>\n" );
+
+  metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Type" ) + QStringLiteral( "</td><td>" ) + domain->typeName() + QStringLiteral( "</td></tr>\n" );
+
+  metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Field type" ) + QStringLiteral( "</td><td>" );
+  switch ( domain->fieldType() )
```

Comment:
It would be nice to have something in GUI utils to handles this: there is a risk of inconsistencies when the same type to string conversion is done in other parts of the code.

Reply:
Ok, done. I've made a new QgsVariantUtils class which contains a static method for this (it's in core, since the providers and various models also need it)

Reply:
@nyalldawson thanks!